api/src/main/java/org/commonjava/maven/galley/io/TransferDecoratorPipeline.java

```diff
@@ -136,7 +136,11 @@ public void decorateDelete( final Transfer transfer, final EventMetadata metadat
         String[] result = listing;
         for ( TransferDecorator decorator : decorators )
         {
-            result = decorator.decorateListing( transfer, listing, metadata );
+            String[] decorated = decorator.decorateListing( transfer, listing, metadata );
+            // Only decorated listing changed need to override the result.
+            if(!Arrays.equals( listing, decorated )){
```

Comment:
The if statement seems not necessary. if the decorated is equals to listing, the assignment is still valid. 
btw, during the for loop, it use an immutable listing. I wonder should here use result instead of listing? i.e., 
String[] decorated = decorator.decorateListing( transfer, result, metadata );
in case the result changed.

Reply:
This if statement is really necessary, because all other Decorator is using default decorateListing from AbstractTransferDecorator, which just simply returned the original listing. The only one did the changed work is ContentsFilterTransferDecorator, but its result is override by other ones, which actually did not change the result. I don't think this is a correct action for this decorateListing method, and from the local testing work, it proved my thoughts: the ContentsFilterDecorator.decorateListing result is override by other ones which did not change listing, which is not the correct case for issue NOS-1434.
So we should only consider the decorated result which actually give changes to the listing result here. And for advance, maybe we still need to think more about this decorateListing logic: is the simple override logic is correct here for the content listing logic?

Reply:
Can we use 'result' in the param instead of 'listing'? i.e.,
result = decorator.decorateListing( transfer, result, metadata );
After ContentsFilterTransferDecorator changed the 'result', the rest of these decorators use the updated 'result' as the param and return it.

Reply:
Maybe this decorateListing is not so accurate from semantic level in this TransferDecoratorPipeline, because if there are more than one Decorator, the following ones may override the result of the previous ones, which are not correct. (Even we use the previous result to do the following logic).
But here I agree to use the result instead but not original listing, because it will not break the logic. So will change.

Reply:
Maybe this decorateListing is not so accurate from semantic level in this TransferDecoratorPipeline, because if there are more than one Decorator, the following ones may override the result of the previous ones, which are not correct. (Even we use the previous result to do the following logic).
But here I agree to use the result instead but not original listing, because it will not break the logic. So will change.

Reply:
Maybe this decorateListing is not so accurate from semantic level in this TransferDecoratorPipeline, because if there are more than one Decorator, the following ones may override the result of the previous ones, which are not correct. (Even we use the previous result to do the following logic).
But here I agree to use the result instead but not original listing, because it will not break the logic. So will change.

Reply:
Maybe this decorateListing is not so accurate from semantic level in this TransferDecoratorPipeline, because if there are more than one Decorator, the following ones may override the result of the previous ones, which are not correct. (Even we use the previous result to do the following logic).
But here I agree to use the result instead but not original listing, because it will not break the logic. So will change.

Reply:
Maybe this decorateListing is not so accurate from semantic level in this TransferDecoratorPipeline, because if there are more than one Decorator, the following ones may override the result of the previous ones, which are not correct. (Even we use the previous result to do the following logic).
But here I agree to use the result instead but not original listing, because it will not break the logic. So will change.

Reply:
Maybe this decorateListing is not so accurate from semantic level in this TransferDecoratorPipeline, because if there are more than one Decorator, the following ones may override the result of the previous ones, which are not correct. (Even we use the previous result to do the following logic).
But here I agree to use the result instead but not original listing, because it will not break the logic. So will change.

Reply:
Maybe this decorateListing is not so accurate from semantic level in this TransferDecoratorPipeline, because if there are more than one Decorator, the following ones may override the result of the previous ones, which are not correct. (Even we use the previous result to do the following logic).
But here I agree to use the result instead but not original listing, because it will not break the logic. So will change.

Reply:
Maybe this decorateListing is not so accurate from semantic level in this TransferDecoratorPipeline, because if there are more than one Decorator, the following ones may override the result of the previous ones, which are not correct. (Even we use the previous result to do the following logic).
But here I agree to use the result instead but not original listing, because it will not break the logic. So will change.

Reply:
Maybe this decorateListing is not so accurate from semantic level in this TransferDecoratorPipeline, because if there are more than one Decorator, the following ones may override the result of the previous ones, which are not correct. (Even we use the previous result to do the following logic).
But here I agree to use the result instead but not original listing, because it will not break the logic. So will change.

Reply:
So I'm not sure if this decorateListing is correct from semantic level: the following decorators' result can still override the previous ones. 
But in current context, I agree to use result instead as it will not break the ContentsFilterDecorator's result. So will change.